src/System.Text.RegularExpressions/tests/Regex.EscapeUnescape.Tests.cs

```diff
@@ -35,6 +41,11 @@ public void Escape_NullString_ThrowsArgumentNullException()
         public void Unescape(string str, string expected)
         {
             Assert.Equal(expected, Regex.Unescape(str));
+
```

Comment:
I suggest breaking the span variations into a separate unit test in each case. This is two tests in one..

Reply:
We follow a similar pattern for string tests where we have both string and span tests in one unit test to re-enforce that the behavior is identical and we have a single place to edit the test cases.

https://github.com/dotnet/corefx/blob/51e1541da82253ab7d10a7c00a2fd93b8e41c341/src/Common/tests/Tests/System/StringTests.cs#L207

We could consider saving the inline data as `IEnumerable<object[]>` and use `MemberData` instead of `InlineData` (to avoid having to duplicate the test cases).

Should we make an exception here and deviate from the pattern, or change string tests as well? Having them together also serves the purpose of highlighting how user code might differ (if at all) between string/span. Thoughts?

cc @tarekgh

Reply:
I think we should follow the string/span test cases and have both in same place for the reasons that @ahsonkhan mentioned

Reply:
Fine